rpcserver.go

```diff
@@ -8794,35 +8792,57 @@ func calculateAssetMaxAmount(ctx context.Context, priceOracle rfq.PriceOracle,
 // quote.
 func notifyUserForPaymentQuotes(
 	stream tchrpc.TaprootAssetChannels_SendPaymentServer,
-	quote *rfqrpc.PeerAcceptedSellQuote, payHash string,
+	quotes []*rfqrpc.PeerAcceptedSellQuote, payHash string,
 	peerPubKey *route.Vertex) error {
 
-	var err error
+	var (
+		err        error
+		firstQuote *rfqrpc.PeerAcceptedSellQuote
+	)
 
-	err = stream.Send(&tchrpc.SendPaymentResponse{
-		Result: &tchrpc.SendPaymentResponse_AcceptedSellOrder{
-			AcceptedSellOrder: quote,
-		},
-	})
-	if err != nil {
-		return err
-	}
+	for _, quote := range quotes {
+		if firstQuote == nil {
+			firstQuote = quote
+		}
 
-	// Unmarshall the accepted quote's asset rate.
-	assetRate, err := rpcutils.UnmarshalRfqFixedPoint(
-		quote.BidAssetRate,
-	)
-	if err != nil {
-		rpcsLog.Errorf("error unmarshalling asset "+
-			"rate for payment %v: %w", payHash, err)
+		// Unmarshall the accepted quote's asset rate.
+		assetRate, err := rpcutils.UnmarshalRfqFixedPoint(
+			quote.BidAssetRate,
+		)
+		if err != nil {
+			rpcsLog.Errorf("error unmarshalling asset rate for "+
+				"payment %v: %w", payHash, err)
+		}
+
+		// Log the negotiated quote for debugging purposes.
+		rpcsLog.Infof("Payment %v got quote for %v asset units at %v "+
+			" asset/BTC from peer %x with SCID %d", payHash,
+			quote.AssetAmount, assetRate, peerPubKey, quote.Scid)
 	}
 
-	rpcsLog.Infof("Payment %v got quote for %v asset "+
-		"units at %v asset/BTC from peer %x with "+
-		"SCID %d", payHash, quote.AssetAmount,
-		assetRate, peerPubKey, quote.Scid)
+	// Let's send the first quote as a single quote stream response. This is
+	// done to maintain compatibility with old behavior, where a single
+	// quote was returned in the response stream.
+	if firstQuote != nil {
+		err = stream.Send(&tchrpc.SendPaymentResponse{
+			Result: &tchrpc.SendPaymentResponse_AcceptedSellOrder{
+				AcceptedSellOrder: firstQuote,
+			},
+		})
+		if err != nil {
+			return err
+		}
+	}
 
-	return nil
+	// We now stream the full array of negotiated quotes to the response
+	// stream.
+	return stream.Send(&tchrpc.SendPaymentResponse{
```

Comment:
:+1: looks good. This will require a small adjustment in `litcli` as well, just as a reminder.